app/src/tests/lib/download.test.ts

```diff
@@ -0,0 +1,94 @@
+const mockAuthHeader = jest.fn();
+jest.mock("../../lib/auth/getAuthHeader", () => ({
+    getAuthHeader: () => mockAuthHeader()
+}));
+
+import {downloadFileUri} from "../../msw/handlers/downloadFileHandlers";
+import {download} from "../../lib/download";
+import {mockFileBlob} from "../mocks";
+import {server} from "../../msw/server";
+import {rest} from "msw";
+
+const url = `${downloadFileUri}?filename=test.txt`;
+
+describe("download", () => {
+    const fakeObjectUrl = "fakeObjectUrl";
+    const realFetch = window.fetch;
```

Comment:
shouldnt need to mock fetch... can use msw handler to intercept request.. msw purpose is so we can avoid this :smile:

Reply:
I want to spy on  it though... :mag:

Reply:
Unless msw has some spy magic I don't know about?

Reply:
what exactly are you spying on? is the direct spyon fetch needed for the test? looks like it line 48 is the assertion... unsure if that is needed since you are already checking lots of assertions afterwards?

Reply:
I think I wanted to use the spy to check the header had been sent correctly, and then I needed to restore the real fetch for subsequent fetches to invoke msw. But actually I could probably configure msw to only kick in when the correct header is provided, to return the mock blob, and that would confirm that the header had been sent.. I'll give that a try.